Double Validation can be too intrusive

boutinb opened the pull request:
When 2 DoubleFields determine the min and max of each other (to simulate a range), if the user changes the value of one field, the min can temporary exceed the max of one DoubleField, though the user did not finish editing the value.. As long a user did not finish editing a value, no error messages should be displayed.

Fixes https://github.com/jasp-stats/jasp-test-release/issues/2170

boutinb committed:
Double Validation can be too intrusive

When 2 DoubleFields determine the min and max of each other (to simulate a range), if the user changes the value of one field, the min can temporary exceed the max of one DoubleField, though the user did not finish editing the value..
As long a user did not finish editing a value, no error messages should be displayed.

Kucharssim reviewed (commented):
It is improved; I don't get stuck in an infinite loop, but it is still a bit intrusive (both validation errors fire up at the same time and I need to fix the value problem in one of the fields and then manually close both messages)

JorisGoosen commented:
mybe @boutinb can add a `editing` property to textfield so you can set the min and mx to infinity while they are edited?

boutinb committed:
Make difference between transientValue and value in Text-Double-Integer Field

transientValue property is the value the user is typing, value property is the value when the user has finished editing.
transientValue property should be used in the onEditingFinished slots in QML.

boutinb commented:
I have defined the `displayValue` property which is the value the user is typing.
The `value` property is then the value when the user has finished editing.
`displayValue` property should be used only in the onEditingFinished slots in QML (which should not be used in the analyses).
So when in a QML, the `value` of a DoubleField or IntegerField is used in a property of another control,  this value won't be updated each time the user is typing, but only when the user has finished editing.
Also this makes the handling of this min - max constraints easier: when modifying the min or max, the constraint is checked only when `value` is changed, so only when the user has finished typing.
I have also add a Min Max test in jaspTestModule

boutinb committed:
Replace transientValue by displayValue

boutinb committed:
Foutje

JorisGoosen commented on QMLComponents/components/JASP/Controls/TextField.qml:
```diff
@@ -33,8 +33,8 @@ TextInputBase
 	property alias	text:				textField.label
-	property alias	value:				control.text
-	property string lastValidValue:		defaultValue
+	property alias	displayValue:		control.text
```
```suggestion
	property alias	displayValue:		control.text	///< Only use in onEditingFinished!
```

JorisGoosen committed:
Add comment on displayValue

JorisGoosen committed:
Improve comments
